src/include/api.h

```diff
@@ -63,10 +93,26 @@
     __oldname = (s)->name;                                                                   \
     ++(s)->api_call_counter;                                                                 \
     (s)->dhandle = (dh);                                                                     \
-    (s)->name = (s)->lastop = #struct_name "." #func_name
-#define API_SESSION_POP(s)  \
-    (s)->dhandle = __olddh; \
-    (s)->name = __oldname;  \
+    (s)->name = (s)->lastop = #struct_name "." #func_name;                                   \
+    if ((s)->id != 0 && (s)->api_call_counter == 1) {                                        \
```

Comment:
What's the deal with checking `api_call_counter == 1`? I don't understand what this does or the interaction with the new counters.

Reply:
Oh, I think I get it now - is it for handling reentrancy?

Reply:
Yes - that's right. It pre-existed this change and avoids double counting on re-entrance (which is common)